### PR TITLE
Catch exceptions

### PR DESCRIPTION
**Pull request by ycanardeau**
#616

### CODE REVIEW

File: VocaDbWeb/Controllers/UserController.cs [hunk 1]
```diff
@@ -597,10 +597,23 @@ public ActionResult Edit(int id)
 
 			PermissionContext.VerifyPermission(PermissionToken.ManageUserPermissions);
 
+			model.OldName = Service.GetUserWithPermissions(model.Id).Name;
```
Comment (ycanardeau) [inline VocaDbWeb/Controllers/UserController.cs:600]: `model.Name` may be replaced by the user. So we need to copy of that.

File: VocaDbWeb/Models/UserModels.cs [hunk 2]
```diff
@@ -308,8 +308,11 @@ public UserEdit(UserWithPermissionsContract contract)
 
 		public int Id { get; set; }
 
+		[Required(ErrorMessageResourceType = typeof(ViewRes.User.CreateStrings), ErrorMessageResourceName = "UsernameIsRequired")]
 		public string Name { get; set; }
 
+		public string OldName { get; set; }
```
Comment (riipah) [inline VocaDbWeb/Models/UserModels.cs:314]: Maybe add a comment here

**Comment by riipah** [comment]
Thanks again. I would make a few small changes. Overall it's fine.

File: VocaDbModel/Database/Queries/UserQueries.cs [hunk 3]
```diff
@@ -1573,7 +1573,15 @@ class CachedUserStats {
 
 				user.Active = contract.Active;
 				user.Email = contract.Email;
-				user.NormalizedEmail = !string.IsNullOrEmpty(contract.Email) ? MailAddressNormalizer.Normalize(contract.Email) : string.Empty;
+
+				if (!string.IsNullOrEmpty(contract.Email)) {
+					ValidateEmail(contract.Email);
+
+					user.NormalizedEmail = MailAddressNormalizer.Normalize(contract.Email);
```
[the remark this answers is not in the document]
Comment (ycanardeau) [reply VocaDbModel/Database/Queries/UserQueries.cs:1580]: Do you mean like this? https://github.com/VocaDB/vocadb/pull/626/commits/3fe23f12188c3aeb1499654ad7bbc9311d21fe3e

**Comment by ycanardeau** [comment]
Thank you for your review!

File: VocaDbModel/Database/Queries/UserQueries.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (riipah) [reply VocaDbModel/Database/Queries/UserQueries.cs:1580]: That's good

**Comment by riipah** [comment]
Thanks, looks fine now!

**Pull request merged**